### PR TITLE
:tada: Component swap

### PR DESCRIPTION
**Pull request by pabloalba**
Implements https://tree.taiga.io/project/penpot/us/1331

### CODE REVIEW

File: common/src/app/common/pages/changes_builder.cljc [hunk 1]
```diff
@@ -323,7 +326,9 @@
                   :shapes (->> shapes reverse (mapv :id))}
 
            (some? index)
-           (assoc :index index))
+           (assoc :index index)
+           (:component-swap options)
+           (assoc :component-swap true))
```
Comment (hirunatan) [inline common/src/app/common/pages/changes_builder.cljc:331]: If you add a property to a change type, you must add it to the change spec, at `common/pages/changes' .

File: common/src/app/common/pages/helpers.cljc [hunk 2]
```diff
@@ -531,6 +531,14 @@
                    (merge-path other-path item))))
         [other-path last-item false]))))
 
+(defn prev-path
```
Comment (hirunatan) [inline common/src/app/common/pages/helpers.cljc]: The name seems a bit confusing. Perhaps `butlast-path`?

File: frontend/resources/styles/main/partials/sidebar-element-options.scss [hunk 3]
```diff
@@ -585,6 +597,26 @@
       }
     }
   }
+
+  &.copy {
```
Comment (hirunatan) [inline frontend/resources/styles/main/partials/sidebar-element-options.scss:601]: question: does this work in new design system?

File: frontend/src/app/main/data/workspace/groups.cljs [hunk 4]
```diff
@@ -101,7 +101,7 @@
 
         ;; Shapes that are in a component, but are not root, must be detached,
         ;; because they will be now children of a non instance group.
-        shapes-to-detach (filter ctk/in-component-copy-not-root? shapes)
+        shapes-to-detach (filter ctk/in-component-copy-not-head? shapes)
```
Comment (hirunatan) [inline frontend/src/app/main/data/workspace/groups.cljs:105]: I think shapes inside a copy should not be allowed to be grouped?

File: frontend/resources/styles/main/partials/sidebar-element-options.scss (hunk 3 above)
Comment (pabloalba) [reply frontend/resources/styles/main/partials/sidebar-element-options.scss:601]: I've talked to Eva, and she said that she prefers to refactor it later

File: frontend/src/app/main/data/workspace/groups.cljs (hunk 4 above)
Comment (pabloalba) [reply frontend/src/app/main/data/workspace/groups.cljs:105]: Oh, true. And ungrouped, masked, and transformed to boolean! Good catch

**Pull request merged**